### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by Spikeysanju**

### CODE REVIEW

File: app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt [hunk 1]
```diff
@@ -226,15 +211,15 @@ class ArticlesFragment : Fragment(R.layout.fragment_articles) {
     }
 
     private fun onConnectivityAvailable() {
+
         binding.run {
-            textNetworkStatus.text = getString(R.string.text_connectivity)
+            textNetworkStatus.apply {
```
Comment (ch8n) [inline app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt:216]: Hi nesting scoped extension or not considered good practice,  consider doing this
```Kotlin
fun onConnectivityAvailable() = with(binding){
// do stuff
}
```

File: app/src/main/java/www/thecodemonks/techbytes/ui/base/BaseFragment.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = getViewBinding(inflater, container)
+        return binding.root
+    }
+
+    protected abstract fun getViewBinding(inflater: LayoutInflater, container: ViewGroup?): VB
+
+    fun toast(message: String) {
```
Comment (ch8n) [inline app/src/main/java/www/thecodemonks/techbytes/ui/base/BaseFragment.kt:57]: make toast as extension its required under both activities, fragments and sometimes in views

File: app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt (hunk 1 above)
Comment (ch8n) [reply app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt:216]: checkout why? https://www.youtube.com/watch?v=YeqGfKmJM_g&t=4s&ab_channel=JetBrainsTV

Comment (Spikeysanju) [reply app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt:216]: Hmm gotcha.. it looks interesting 💡

Comment (Spikeysanju) [reply app/src/main/java/www/thecodemonks/techbytes/ui/articles/ArticlesFragment.kt]: Fixed 👍

**Pull request merged**